src/bio.c

```diff
@@ -226,34 +234,40 @@ void *bioProcessBackgroundJobs(void *arg) {
         pthread_mutex_unlock(&bio_mutex[worker]);
 
         /* Process the job accordingly to its type. */
-        if (job->header.type == BIO_CLOSE_FILE) {
-            if (job->fd_args.need_fsync) {
-                redis_fsync(job->fd_args.fd);
-            }
-            close(job->fd_args.fd);
-        } else if (job->header.type == BIO_AOF_FSYNC) {
-            /* The fd may be closed by main thread and reused for another
-             * socket, pipe, or file. We just ignore these errno because
-             * aof fsync did not really fail. */
-            if (redis_fsync(job->fd_args.fd) == -1 &&
-                errno != EBADF && errno != EINVAL)
-            {
-                int last_status;
-                atomicGet(server.aof_bio_fsync_status,last_status);
-                atomicSet(server.aof_bio_fsync_status,C_ERR);
-                atomicSet(server.aof_bio_fsync_errno,errno);
-                if (last_status == C_OK) {
-                    serverLog(LL_WARNING,
-                        "Fail to fsync the AOF file: %s",strerror(errno));
+        switch (job->header.type)
+        {
+            case BIO_CLOSE_FILE:
```

Comment:
i'd much rather avoid that refactor from if-else chain to switch-case.
causes a large diff that's hard to review and causes more merge conflicts.

Reply:
Sure, I should be able to keep it the old way. Although theoretically using a `switch` should be more efficient since it creates a lookup table... :slightly_smiling_face:

Reply:
the compiler can easily convert an if-else chain to a jump table and vice versa, it doesn't really matter these days.